tests/features.rs

```diff
@@ -960,4 +960,48 @@ fn dep_feature_in_cmd_line() {
                 execs().with_status(101).with_stderr("\
 [ERROR] feature names may not contain slashes: `bar/some-feat`
 "));
+
+#[test]
+fn all_features_flag_enables_all_features() {
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+
+            [features]
+            foo = []
+            bar = ["baz"]
```

Comment:
This dependency on `"baz"` may want to be removed because otherwise activating `bar` is the same as activating `baz` (e.g. `--all-features` should activate optional dependencies that aren't listed in `[features]`)